src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -113,16 +113,27 @@ public bool TailCall
         /// <returns>A delegate containing the compiled version of the lambda.</returns>
         public Delegate Compile()
         {
-#if FEATURE_CORECLR
+#if FEATURE_COMPILE
             return Compiler.LambdaCompiler.Compile(this);
 #else
             return new System.Linq.Expressions.Interpreter.LightCompiler().CompileTop(this).CreateDelegate();
 #endif 
         }
 
-#if FEATURE_CORECLR
+#if FEATURE_INTERPRET
+        /// <summary>
+        /// Produces a delegate that represents the lambda expression.
+        /// </summary>
+        /// <returns>A delegate containing the compiled version of the lambda.</returns>
+        public Delegate Interpret()
+        {
```

Comment:
Compile(bool preferInterpretation)   should use interpreter when preferInterpretation == true and interpreter is available (I see no reasons for it to be unavailable, but who knows..)

When preferInterpretation == false, it behaves like the old Compile() - it will use compiler if available and interpreter otherwise.